examples/chants/components/chants/chant.tsx

```diff
@@ -0,0 +1,305 @@
+        state.headerBackground.set("");
+        return;
+      }
+
+      // if the chant is the same as the previous chant, increment the count
+      if (state.emoji.get()?.emojiId === chant.emoji!.emojiId) {
+        return state.count.set((prev) => prev + 1);
+      }
+
+      // otherwise, reset the chant
+      state.emoji.set(chant.emoji!);
+      state.emoji_src.set(chant.emoji!.image.thumbnails[0].url);
+      state.count.set(1);
+      state.show.set(false);
+      state.animation.set("");
+      state.pillBackground.set("");
+      state.headerBackground.set("");
+    });
+  }, []);
+
+  // listen for new chat messages
+  useEffect(() => {
+    const onEmit = (matches: MatchedMessage[]) => {
+      for (const match of matches) {
+        const e = new CustomEvent("message", { detail: match });
+        target.dispatchEvent(e);
+      }
+    };
+
+    jumper.call(
+      "layout.listenForElements",
+      {
+        listenElementLayoutConfigSteps: [
+          { action: "querySelector", value: "#chatframe" },
+          { action: "getIframeDocument" },
+          { action: "querySelector", value: "#item-scroller #items" },
+        ],
+        observerConfig: { childList: true, subtree: true },
+        targetQuerySelector: "yt-live-chat-text-message-renderer",
+        shouldCleanupMutatedElements: true,
+      },
+      onEmit,
+    );
+  }, []);
+
+  // monitors `state.count` and `state.show` and
+  // controls the animations and effects as the count changes
+  useObserve(() => {
+    state.animation.set("");
+
+    const count = state.count.get();
+
+    new Promise((resolve) => setTimeout(resolve, 10)).then(() =>
+      state.animation.set("pop 0.3s ease-in-out")
+    );
+
+    // TODO - move the counts to constants
+    if (!state.show.get() && count >= 2) state.show.set(true);
+    if (count >= 3) {
+      return state.headerBackground.set(
+        "linear-gradient(124deg, #ff2400, #e81d1d, #e8b71d, #e3e81d, #1de840, #1ddde8, #2b1de8, #dd00f3, #dd00f3)",
+      );
+    }
+
+    if (count >= 2) {
+      return state.pillBackground.set(
+        "linear-gradient(124deg, #ff2400, #e81d1d, #e8b71d, #e3e81d, #1de840, #1ddde8, #2b1de8, #dd00f3, #dd00f3)",
+      );
+    }
+
+    if (count <= 1) return;
+  });
+
+  const emojiSrc = useSelector(state.emoji_src);
+  const show = useSelector(state.show);
+  const background = useSelector(state.pillBackground);
+  const emoji = useSelector(state.emoji);
+
+  const onClick = () => {
+    jumper.call("layout.applyLayoutConfigSteps", {
+      layoutConfigSteps: [
+        {
+          action: "querySelector",
+          value: "yt-live-chat-text-input-field-renderer",
+        },
+        { action: "youtubeSetInputText", value: `${emoji?.shortcuts[0]}` },
+      ],
+    });
+  };
+  return show
+    ? (
+      <div className="c-chants">
+        <div
+          className="chant-container"
+          onClick={onClick}
+          style={{
+            width: "80px",
```

Comment:
imo we should use css vs definiing styles inline (so putting all of these except background in the css file .chant-container